compositor_pipeline/src/pipeline/whip_whep/whip_handlers/create_whip_session.rs

```diff
@@ -0,0 +1,163 @@
+        let track_kind = track.kind();
+        let state_clone = state.clone();
+        let input_id_clone = input_id.clone();
+        let depayloader_clone = input_components.depayloader.clone();
+        let sender = match track_kind {
+            RTPCodecType::Video => input_components.video_sender.clone(),
+            RTPCodecType::Audio => input_components.audio_sender.clone(),
+            _ => {
+                debug!("RTPCodecType not supported!");
+                None
+            }
+        };
+
+        //tokio::spawn is necessary to concurrently process audio and video track
+        Box::pin(async {
+            if let Some(sender) = sender {
+                tokio::spawn(async move {
+                    process_track_stream(
+                        track,
+                        state_clone,
+                        input_id_clone,
+                        depayloader_clone,
+                        sender,
+                    )
+                    .await;
+                });
+            }
+        })
+    }));
+
+    peer_connection.on_ice_connection_state_change(Box::new(move |state| {
+        info!("ICE connection state changed: {state:?}");
+        Box::pin(async {})
+    }));
+
+    let description = RTCSessionDescription::offer(offer)?;
+
+    peer_connection.set_remote_description(description).await?;
+    let answer = peer_connection.create_answer(None).await?;
+    peer_connection.set_local_description(answer).await?;
+    gather_ice_candidates_for_one_second(peer_connection.clone()).await;
+
+    let Some(sdp) = peer_connection.local_description().await else {
+        return Err(WhipServerError::InternalError(
+            "Local description is not set, cannot read it".to_string(),
+        ));
+    };
+    debug!("Sending SDP answer: {sdp:?}");
+
+    let body = Body::from(sdp.sdp.to_string());
+    let response = Response::builder()
+        .status(StatusCode::CREATED)
+        .header("Content-Type", "application/sdp")
+        .header("Access-Control-Expose-Headers", "Location")
+        .header("Location", format!("/session/{}", encode(&id)))
+        .body(body)?;
+    Ok(response)
+}
+
+pub fn validate_sdp_content_type(headers: &HeaderMap) -> Result<(), WhipServerError> {
+    if let Some(content_type) = headers.get("Content-Type") {
+        if content_type.as_bytes() != b"application/sdp" {
+            return Err(WhipServerError::InternalError(
+                "Invalid Content-Type".to_string(),
+            ));
+        }
+    } else {
+        return Err(WhipServerError::BadRequest(
+            "Missing Content-Type header".to_string(),
+        ));
+    }
+    Ok(())
+}
+
+pub async fn gather_ice_candidates_for_one_second(peer_connection: Arc<RTCPeerConnection>) {
+    let (sender, mut receiver) = watch::channel(peer_connection.ice_gathering_state());
+
+    peer_connection.on_ice_gathering_state_change(Box::new(move |gatherer_state| {
+        let gathering_state = match gatherer_state {
```

Comment:
why are you translating from one type to the other if the only thing you do in the end is compare to RTCIceGatheringState::Complete ?